openedx_tagging/core/tagging/models.py

```diff
@@ -1,39 +1,428 @@
 """ Tagging app data models """
+from typing import List, Type
+
 from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from openedx_learning.lib.fields import MultiCollationTextField, case_insensitive_char_field
```

Comment:
Should we move the fields stuff out of `openedx_learning` and into something else? I'm just worried about circular dependencies here, since my understanding is that eventually there will something in `openedx_learning.contrib` that imports from `openedx_tagging` to link tags to the content data models?

Reply:
Hmm.. you're right.. maybe they'd be better in https://github.com/openedx/edx-django-utils/ ?

Reply:
Great plan, thank you @ormsbee . Should I codify this in an ADR in my next PR?

Reply:
@ormsbee  These fields are working great on MySQL, but I can't use them in sqlite in the LMS tests :(

Running the tests with the branch from https://github.com/openedx/edx-platform/pull/32518 is throwing these errors when I run them on my [devstack](https://github.com/openedx/devstack)'s `lms-shell`:

```
root@lms:/edx/app/edxapp/edx-platform# pytest openedx/features/content_tagging/
<snip>
../venvs/edxapp/lib/python3.8/site-packages/pytest_django/plugin.py:490:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../venvs/edxapp/lib/python3.8/site-packages/pytest_django/fixtures.py:122: in django_db_setup
    db_cfg = setup_databases(
../venvs/edxapp/lib/python3.8/site-packages/django/test/utils.py:179: in setup_databases
    connection.creation.create_test_db(
../venvs/edxapp/lib/python3.8/site-packages/django/db/backends/base/creation.py:74: in create_test_db
    call_command(
../venvs/edxapp/lib/python3.8/site-packages/django/core/management/__init__.py:181: in call_command
    return command.execute(*args, **defaults)
../venvs/edxapp/lib/python3.8/site-packages/django/core/management/base.py:398: in execute
    output = self.handle(*args, **options)
../venvs/edxapp/lib/python3.8/site-packages/pytest_django/fixtures.py:276: in handle
    return super().handle(*args, **kwargs)
../venvs/edxapp/lib/python3.8/site-packages/django/core/management/base.py:89: in wrapped
    res = handle_func(*args, **kwargs)
../venvs/edxapp/lib/python3.8/site-packages/django/core/management/commands/migrate.py:262: in handle
    post_migrate_apps.render_multiple([
../venvs/edxapp/lib/python3.8/site-packages/django/core/management/commands/migrate.py:263: in <listcomp>
    ModelState.from_model(apps.get_model(*model)) for model in model_keys
../venvs/edxapp/lib/python3.8/site-packages/django/db/migrations/state.py:407: in from_model
    fields.append((name, field.clone()))
../venvs/edxapp/lib/python3.8/site-packages/django/db/models/fields/__init__.py:513: in clone
    return self.__class__(*args, **kwargs)
<snip>
ERROR openedx/features/content_tagging/tests/test_api.py::TestAPIContentTaxonomy::test_content_taxonomy_enabled_for_org1 - ValueError: Cannot use db_collation with MultiCollationCharField. Please use a db_collations dict instead.
```

And I found your post related to this on https://forum.djangoproject.com/t/supporting-collation-settings-on-multiple-database-backends/12561/, but I'm unclear how (or if) I can work around this.

Should I just roll these back to being normal, default-collation CharField/TextField until we work out a solution for both sqlite and MySQL?

Reply:
If I remove the check ([a1a90d2](https://github.com/openedx/openedx-learning/pull/57/commits/a1a90d2689dd1537e2cc87544945d3abe0362e17)) then my tests on https://github.com/openedx/edx-platform/pull/32518 pass, cf https://github.com/openedx/edx-platform/actions/runs/5383371513/jobs/9770044365

Are you ok with merging this without that check?

Reply:
Yes. I'm going to look through it later to try to figure out why it did this, but I'm fine with removing it for now. Thank you.